deck.py

```diff
@@ -51,37 +39,38 @@ def is_empty(self):
     def create_new_deck_with_played_cards(self, played_cards):
         """
         It takes the played cards and create a new deck, then it returns
-        the self.get_deck() method that shuffles the deck and returns it
+        an instance of itself with the new deck
         """
         print("Creating new deck with the cards played...\n")
         self.cards = played_cards
-        return self.get_deck()
+        self.shuffle_deck()
+        return _Deck(self.cards)
 
 import unittest
 
 class TestDeck(unittest.TestCase):
     def test_empty_deck(self):
-        deck = Deck([])
+        deck = _Deck([])
         self.assertListEqual(deck.cards, [])
 
     def test_deck_with_cards(self):
-        deck = Deck([1,2,3,4,5])
+        deck = _Deck([1,2,3,4,5])
         self.assertListEqual(deck.cards, [1,2,3,4,5])
 
     def test_shuffled_deck_other_than_base_deck(self):
-        deck = Deck([1,2,3,4,5])
+        deck = _Deck([1,2,3,4,5])
         cards_before = deck.cards.copy()
         deck.shuffle_deck()
         self.assertNotEqual(deck.cards, cards_before)
 
     def test_draw_card(self):
-        deck = Deck([1,2,3,4,5])
+        deck = _Deck([1,2,3,4,5])
         card = deck.draw_card()
         self.assertEqual(card, 5)
         self.assertEqual(len(deck.cards), 4)
 
 if __name__ == "__main__":
     unittest.main()
     
-class NoMoreCards(Exception):
+class NoMoreCardsError(Exception):
```

Comment:
Thats good change. In case of creating Exception its better to add """docstring""" in place of pass so you can document when this exception is raised :)